src/NQueen.java

```diff
@@ -0,0 +1,132 @@
+import java.util.Scanner;
+public class NQueen 
+{
+	public static void main(String args[]) throws Exception
+	{
+		Scanner scanner =new Scanner(System.in);
+		NQueen obj=new NQueen();
+		System.out.println("Enter the dimension of matrix : ");
+		int dimensionOfMatrix=scanner.nextInt();
+		scanner.close();
+		int result[][]=obj.createMatrix(dimensionOfMatrix);
+		NQueen.printResult(result);
+	}
+	
+	/**
+	 * @param dimensionOfMatrix
+	 * @return solution of N queen in 2D array
+	 * @throws Exception custom exception for n<=3
+	 */
+	public int[][] createMatrix(int dimensionOfMatrix)throws Exception
+	{
+		if(dimensionOfMatrix<=3)
+		{
+			throw new Exception("No Solution exist for N smaller than 3");
+		}
+		int board[][]=new int[dimensionOfMatrix][dimensionOfMatrix];
+		for(int loop_var=0;loop_var<dimensionOfMatrix;loop_var++)
+		{
+			for( int inner_loop_var=0;inner_loop_var<dimensionOfMatrix;inner_loop_var++)
+			{
+				board[loop_var][inner_loop_var]=0;
+			}
+		}
+		Boolean result = nQueen(board , 0, dimensionOfMatrix);
+		if(result==true)
+		{
+			return board;
+		}
+		else
+		{
+			System.out.println("No sulution exist for provided dimensions");
+			System.exit(-1);
+		}
+		return null;
+	}
+	/**
+	 * @param board
+	 * Print the solution
+	 */
+	public static void printResult(int[][] board)
+	{
+		for(int loop_var=0;loop_var<board.length;loop_var++)
+		{
+			for( int inner_loop_var=0;inner_loop_var<board.length;inner_loop_var++)
+			{
+				System.out.print(board[loop_var][inner_loop_var]+" ");
+			}
+			System.out.println();
+		}
+	}
+	/**
+	 * @param board 
+	 * @param row
+	 * @param col
+	 * @param dimensionOfMatrix
+	 * @return return true if queen is safe to place
+	 */
+	boolean isSafe(int board[][], int col, int row, int dimensionOfMatrix)
+    {
+        int i, j;
+        // Vertical checking for other queens above only
+        for (i = 0; i < row; i++)
+            {
+        		if (board[col][i] == 1)
+        		{
+        			return false;
+        		}
+        	}
+        // Left Diagonal checking for other queens above only
+        for (i = col, j = row; i >= 0 && j >= 0; i--, j--)
+            {
+        		if (board[i][j] == 1)
+        		{
+        			return false;
+        		}
+            }
+        
+        // Right Diagonal checking for other queens above only
+        for (i = col, j = row;i < dimensionOfMatrix && j >= 0 ; i++, j--)
+            {
+        		if (board[i][j] == 1)
+        		{
+        			return false;
+        		}
+            }
```

Comment:
You could check both diagonal in a single loop

Reply:
The dimension for row and column will be different for different diagonal.

Reply:
What was the case which could not be handle into a single loop?

Reply:
Optimized the solution for isSafe methods in nQueen problem